include/sensors/core/callback_types.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+namespace sensor_callbacks {
+using MaxMessageBuffer = std::array<uint8_t, 5>;
+
+// NOLINTNEXTLINE(cppcoreguidelines-special-member-functions)
+class SingleRegisterCallback {
```

Comment:
I chose to do a pure virtual function because I honestly didn't know how else to type this callback. The reason for having two overloaded operator functions is so that the i2c task can separately choose to send a CAN message back. This is most relevant for polling a specific i2c bus where you'll want to accumulate the data without actually sending a message back.